base/src/STLRendererSink.cpp

```diff
@@ -0,0 +1,171 @@
+#include<boost/serialization/vector.hpp>
+#include "STLRendererSink.h"
+#include "ApraPoint3f.h"
+
+class Detail
+{
+private:
+	STLRendererSinkProps mProps;
+public:
+	Detail(STLRendererSinkProps& _props)
+	{
+		mProps = _props;
+	}
+
+	~Detail()
+	{
+		LOG_TRACE << "STLRenderer::Detail destructor";
+	}
+
+	void renderMesh(vector<ApraPoint3f>& points3D)
+	{
+		// https://www.kitware.com/a-tour-of-vtk-pointer-classes/
+
+		auto nPoints = points3D.size();
+		vtkNew<vtkPoints> points;
+		for (auto i = 0; i < nPoints; ++i)
+		{
+			auto& pts = points3D[i];
+			points->InsertNextPoint(pts.x, pts.y, pts.z);
+		}
+
+		// define topography using point IDs (indices)
+		vtkNew<vtkCellArray> triangles;
+		for (vtkIdType i = 0; i < nPoints; i += 3)
+		{
+			vtkIdType ids[3] = { i, i + 1, i + 2 };
+			triangles->InsertNextCell(3, ids);
+		}
+
+		LOG_INFO << "nPoints <" << points->GetNumberOfPoints() << ">";
+		LOG_INFO << "nTriangles <" << triangles->GetNumberOfCells() << ">";
+
+		vtkNew<vtkPolyData> polyData;
+		polyData->SetPoints(points);
+		polyData->SetPolys(triangles);
+
+		// vtk pipeline
+		vtkNew<vtkPolyDataMapper> meshMapper;
+		meshMapper->SetInputData(polyData);
+		meshMapper->Update();
+
+		vtkNew<vtkActor> meshActor;
+		meshActor->SetMapper(meshMapper);
+
+		vtkNew<vtkNamedColors> colors;
+		meshActor->GetProperty()->SetDiffuse(0.8);
+		//auto col = colors->GetColor3d("LightSteelBlue").GetData();
+		double meshDiffuseColor[3] = { mProps.meshDiffuseColor[0] / 255.0,
+			mProps.meshDiffuseColor[1] / 255.0,
+			mProps.meshDiffuseColor[2] / 255.0 };
+		meshActor->GetProperty()->SetDiffuseColor(vtkColor3d(meshDiffuseColor[0], meshDiffuseColor[1], meshDiffuseColor[2]).GetData());
+		meshActor->GetProperty()->SetSpecular(mProps.meshSpecularCoefficient);
+		meshActor->GetProperty()->SetSpecularPower(mProps.meshSpecularPower);
+
+		// rendering
```

Comment:
Please reuse window, window interactor and camera. You can use change props to handle changes in window and camera properties.